FREEWARE/v40/moonclock/src/oclock.man
.TH OCLOCK 1 "Release 5" "X Version 11"
.SH NAME
oclock \- display time of day and moon phase
.SH SYNOPSIS
.B oclock
[-option ...]
.SH DESCRIPTION
.I Clock
simply displays the current time on an analog display
.SH STANDARD OPTIONS
.TP 8
.B \-fg \fIforeground color\fB
choose a different color for the both hands and the jewel of the clock
.TP 8
.B \-bg \fIbackground color\fB
choose a different color for the background.
.TP 8
.B \-jewel \fIjewel color\fB
choose a different color for the jewel on the clock.
.TP 8
.B \-minute \fIminute color\fB
choose a different color for the minute hand of the clock.
.TP 8
.B \-hour \fIhour color\fB
choose a different color for the hour hand of the clock.
.TP 8
.B \-backing \fI{ WhenMapped Always NotUseful }\fB
selects an appropriate level of backing store.
.TP 8
.B \-geometry \fIgeometry\fB
define the initial window geometry; see \fIX(1)\fP.
.TP 8
.B \-display \fIdisplay\fB
specify the display to use; see \fIX(1)\fP.
.TP 8
.B \-bd \fIborder color\fB
choose a different color for the window border.
.TP 8
.B \-bw \fIborder width\fB
choose a different width for the window border.  As the Clock widget changes
its border around quite a bit, this is most usefully set to zero.
.TP 8
.B \-noshape
causes the clock to not reshape itself and ancestors to exactly
fit the outline of the clock.
.TP 8
.B \-transparent
causes the clock to consist only of the jewel, the hands, and the border.
.SH MOON OPTIONS
.TP 8
.B \-moonclock
switch oclock into moonclock mode
.TP 8
.B \-moon \fImoon color\fB
choose a different color for the moon
.TP 8
.B \-sky \fIsky color\fB
choose a different color for the sky
.TP 8
.B \-stars \fIstars color\fB
choose a different color for the "stars"
.TP 8
.B \-night \fInight color\fB
choose a different color for the border and
the sky, when the night has begun.
.TP 8
.B \-nightstart \fI 24h hour
by default the night-colors are set at 6pm (18h). 
you can change this time (early morning hackers might
prefer values between 2 and 4 ?)
.TP 8
.B \-timezone \fI +- 24h hour
you can set another timezone; this is useful, if
there are many clocks on your screen (One oclock
for SFO, another for Germany ?)
.SH NOTES RELATING MOON-MODE
Please see the Copyright-Notices in the moon-related files.
These files are copyed from xphoon.
.B
It is not possible to use oclock in transparent and moonclock mode at the same time.
.SH COLORS
If you would like your clock to be viewable in color, include the following 
in the #ifdef COLOR section you read with xrdb:
.sp 1
*customization:                 -color
.sp 1
.br
This will cause oclock to pick up the colors in the app-defaults color
customization file:
/usr/lib/X11/app-defaults/Clock-color.
Below are the default colors:
.sp 1
Clock*Background: lightsteelblue2
.br
Clock*Foreground: violetred1
.br
Clock*BorderColor: white
.br
Clock*hour: yellow
.br
Clock*jewel: yellow
.br
Clock*minute: yellow
.br
Clock*night: dodgerblue3
.br
Clock*moon: yellow
.br
Clock*sky: dodgerblue1
.br
Clock*stars: white
.SH "SEE ALSO"
X(1), X Toolkit documentation
.SH COPYRIGHT
Copyright 1989, Massachusetts Institute of Technology.
.br
See \fIX(1)\fP for a full statement of rights and permissions.
.SH AUTHORS
Keith Packard, MIT X Consortium
.br
moonclock extensions by Gunter Sprenger
.br
gunter@cs.tu-berlin.de



FREEWARE/v30/moonclock/src/oclock.c
/*
 * $XConsortium: oclock.c,v 1.13 91/01/10 21:33:48 gildea Exp $
 *
 * Copyright 1989 Massachusetts Institute of Technology
 *
 * Permission to use, copy, modify, distribute, and sell this software and its
 * documentation for any purpose is hereby granted without fee, provided that
 * the above copyright notice appear in all copies and that both that
 * copyright notice and this permission notice appear in supporting
 * documentation, and that the name of M.I.T. not be used in advertising or
 * publicity pertaining to distribution of the software without specific,
 * written prior permission.  M.I.T. makes no representations about the
 * suitability of this software for any purpose.  It is provided "as is"
 * without express or implied warranty.
 *
 * M.I.T. DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS SOFTWARE, INCLUDING ALL
 * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS, IN NO EVENT SHALL M.I.T.
 * BE LIABLE FOR ANY SPECIAL, INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
 * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
 * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN 
 * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 */

#include <X11/Intrinsic.h>
#include <X11/Xatom.h>
#include <X11/StringDefs.h>
#include <X11/Shell.h>
#include "Clock.h"
#include <stdio.h> 

#include "oclock.bit"
#include "oclmask.bit"

extern void exit();
static void quit();

static XtActionsRec actions[] = {
    {"quit",	quit}
};

static Atom wm_delete_window;

/* Command line options table.  Only resources are entered here...there is a
   pass over the remaining options after XtParseCommand is let loose. */

/* Exit with message describing command line format */

void usage()
{
    fprintf(stderr,
"usage: oclock\n");
    fprintf (stderr, 
"       [-geometry [{width}][x{height}][{+-}{xoff}[{+-}{yoff}]]] [-display [{host}]:[{vs}]]\n");
    fprintf(stderr,
"       [-fg {color}] [-bg {color}] [-bd {color}] [-bw {pixels}]\n");
    fprintf(stderr,
"       [-minute {color}] [-hour {color}] [-jewel {color}] [-night {color}]\n");
    fprintf(stderr, 
"       [-moonclock] [-sky {color}] [-moon { color}] [-stars {color}]\n");    
    fprintf(stderr,
"       [-timezone {Zone Offset}] [-nightstart { Night Offset }]");
    fprintf(stderr,
"       [-backing {backing-store}]\n");
    exit(1);
}

static XrmOptionDescRec options[] = {
{"-fg",		"*Foreground",		XrmoptionSepArg,	NULL},
{"-bg",		"*Background",		XrmoptionSepArg,	NULL},
{"-foreground",	"*Foreground",		XrmoptionSepArg,	NULL},
{"-background",	"*Background",		XrmoptionSepArg,	NULL},
{"-minute",	"*clock.minute",	XrmoptionSepArg,	NULL},
{"-hour",	"*clock.hour",		XrmoptionSepArg,	NULL},
{"-jewel",	"*clock.jewel",		XrmoptionSepArg,	NULL},
{"-timezone",   "*clock.timezone",      XrmoptionSepArg,        NULL},
{"-nightstart", "*clock.nightstart",    XrmoptionSepArg,        NULL},
{"-night",	"*clock.night",		XrmoptionSepArg,	NULL},
{"-sky",        "*clock.sky",           XrmoptionSepArg,	NULL},
{"-moon",       "*clock.moon",          XrmoptionSepArg,	NULL},
{"-stars",      "*clock.stars",         XrmoptionSepArg,	NULL},
{"-backing",	"*clock.backingStore",	XrmoptionSepArg,	NULL},
{"-shape",	"*clock.shapeWindow",	XrmoptionNoArg,		"TRUE"},
{"-noshape",	"*clock.shapeWindow",	XrmoptionNoArg,		"FALSE"},
{"-transparent","*clock.transparent",	XrmoptionNoArg,		"TRUE"},
{"-moonclock",  "*clock.moonclock",     XrmoptionNoArg,		"TRUE"} 
};

#ifdef VMS
int
#else
void
#endif /* VMS */
 main(argc, argv)
    int argc;
    char **argv;
{
    XtAppContext xtcontext;
    Widget toplevel;
    Widget clock;
    Arg arg[2];
    int	i;
    
    toplevel = XtAppInitialize(&xtcontext, "Clock", options, XtNumber(options),
			       &argc, argv, NULL, NULL, 0);

    if (argc != 1) usage();

    XtAppAddActions
	(xtcontext, actions, XtNumber(actions));
    XtOverrideTranslations
	(toplevel, XtParseTranslationTable ("<Message>WM_PROTOCOLS: quit()"));

    i = 0;
    XtSetArg (arg[i], XtNiconPixmap, 
	      XCreateBitmapFromData (XtDisplay(toplevel),
				     XtScreen(toplevel)->root,
				     (char *)oclock_bits, oclock_width, oclock_height));
    i++;
    XtSetArg (arg[i], XtNiconMask,
	      XCreateBitmapFromData (XtDisplay(toplevel),
				     XtScreen(toplevel)->root,
				     (char *)oclmask_bits, oclmask_width, oclmask_height));
    i++;
    XtSetValues (toplevel, arg, i);

    clock = XtCreateManagedWidget ("clock", clockWidgetClass, toplevel, NULL, 0);
    XtRealizeWidget (toplevel);

    wm_delete_window = XInternAtom(XtDisplay(toplevel), "WM_DELETE_WINDOW",
				   False);
    (void) XSetWMProtocols (XtDisplay(toplevel), XtWindow(toplevel),
                            &wm_delete_window, 1);

    XtAppMainLoop(xtcontext);
}

static void quit(w, event, params, num_params)
    Widget w;
    XEvent *event;
    String *params;
    Cardinal *num_params;
{
    if (event->type == ClientMessage && 
	event->xclient.data.l[0] != wm_delete_window) {
	XBell(XtDisplay(w), 0);
    } else {
	XCloseDisplay(XtDisplay(w));
	exit(0);
    }
}
